src/Pusher.php

```diff
@@ -745,4 +751,46 @@ public function notify($interests, $data = array(), $debug = false)
 
         return false;
     }
+
+    /**
+     * Verify that a webhook actually came from Pusher, and marshals them into a Webhook object.
+     *
+     * @param array  $headers an array of headers from the request (for example, from getallheaders())
+     * @param string $body    the body of the request (for example, from file_get_contents('php://input'))
+     *
+     * @return Webhook object with the properties time_ms (an int) and events (an array of event objects)
+     */
+    public function webhook($headers, $body)
+    {
+        $this->ensure_valid_signature($headers, $body);
+        $decoded_events = array();
+        $decoded_json = json_decode($body);
+        $webhookobj = new Webhook();
+        $webhookobj->time_ms = $decoded_json->time_ms;
+        $webhookobj->events = $decoded_json->events;
+
+        return $webhookobj;
+    }
+
+    /**
+     * Verify that a given Pusher Signature is valid.
+     *
+     * @param array  $headers an array of headers from the request (for example, from getallheaders())
+     * @param string $body    the body of the request (for example, from file_get_contents('php://input'))
+     *
+     * @return bool true if signature is correct.
```

Comment:
It no longer returns anything. Should use https://docs.phpdoc.org/references/phpdoc/tags/throws.html instead.

Reply:
True, will fix.